flink-connector/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/DeltaBulkBucketWriter.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.functions.sink.filesystem;
+
+import java.io.IOException;
+
+import org.apache.flink.api.common.serialization.BulkWriter;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.core.fs.RecoverableFsDataOutputStream;
+import org.apache.flink.core.fs.RecoverableWriter;
+import org.apache.flink.util.Preconditions;
+
+/**
+ * This class is provided as a part of workaround for getting actual file size.
```

Comment:
not clear what this class does. the reason it exists does not quite make sense without more information abouut what it does

Reply:
actually it is explained below:
```
Compared to its original version {@link BulkPartWriter} it changes only the return types
for methods {@link this#resumeFrom} and {@link this#openNew} to a custom implementation
of {@link BulkPartWriter} that is {@link DeltaBulkPartWriter}.
 ```
 
So we need this class only to modify the objects being returned from mentioned functions. And the actual workaround is described and implemented in `DeltaBulkPartWriter`.

Another thing is why we need it at all but to understand this one must get familiar with classes belonging to the package `org.apache.flink.streaming.api.functions.sink.filesystem` in Flink (eg. `flink-file-sink-common` module which we are utilizing in out implementation). Those packages/modules/classes are like a generic ones for interacting with file systems in Flink.
Long story short all of the interfaces are build around the concept of classes like `BucketWriter`, `InProgressFileWriter`, `BulkBucketWriter` and so on which are user-facing abstractions. Then during runtime Flink resolves the correct implementations to use like `org.apache.flink.runtime.fs.hdfs.HadoopRecoverableWriter` or `org.apache.flink.core.fs.local.LocalRecoverableWriter` but on a sink-implementation level it's a black box for us.